src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -357,4 +361,345 @@ int ACA_Dataplane_OVS::update_port_state_workitem(const PortState current_PortSt
   return overall_rc;
 }
 
+int ACA_Dataplane_OVS::update_neighbor_state_workitem(NeighborState current_NeighborState,
+                                                      GoalState &parsed_struct,
+                                                      GoalStateOperationReply &gsOperationReply)
+{
+  int rc, overall_rc;
+  struct sockaddr_in sa;
+  string virtual_ip_address;
+  string virtual_mac_address;
+  string host_ip_address;
+  NetworkType found_network_type;
+  uint found_tunnel_id;
+  string found_gateway_mac;
+  string neighbor_host_dvr_mac;
+  bool subnet_info_found = false;
+  ulong culminative_dataplane_programming_time = 0;
+  ulong culminative_network_configuration_time = 0;
+
+  auto operation_start = chrono::steady_clock::now();
+
+  NeighborConfiguration current_NeighborConfiguration =
+          current_NeighborState.configuration();
+
+  assert(current_NeighborConfiguration.format_version() > 0);
+
+  // TODO: need to design the usage of current_NeighborConfiguration.revision_number()
+  assert(current_NeighborConfiguration.revision_number() > 0);
+
+  switch (current_NeighborState.operation_type()) {
+  case OperationType::CREATE:
+
+    if (current_NeighborConfiguration.neighbor_type() == NeighborType::L2 ||
+        current_NeighborConfiguration.neighbor_type() == NeighborType::L3) {
+      try {
+        assert(current_NeighborConfiguration.fixed_ips_size() == 1);
```

Comment:
What would be the challenges to hock up multiple ips to one port? How about multiple ports per VM? Checking to see if we could explore multiple ports to increase our stress number.

Reply:
> What would be the challenges to hock up multiple ips to one port? 

The primary IP is provided to the VM port by DHCP, the second and more IP will need customer to:
1. manually enter the secondary IPs in their OS
2. run some magic script in their OS to get the secondary IPs
3. use a customer OS by the cloud provider which is able to query and accept secondary IPs

> How about multiple ports per VM? Checking to see if we could explore multiple ports to increase our stress number.

It may just work :) Nova will create the second virtual device and attached to second OVS port. From Alcor to ACA point of view, it is simply another port::create message.